examples/features/nse-composition/README.md

```diff
@@ -0,0 +1,313 @@
+# Test NSE composition
+
+This example demonstrates a more complex Network Service, where we chain three passthrough and one ACL Filtering NS endpoints.
+It demonstrates how NSM allows for service composition (chaining).
+It involves a combination of kernel and memif mechanisms, as well as VPP enabled endpoints.
+
+## Requires
+
+Make sure that you have completed steps from [basic](../../basic) or [memory](../../memory) setup.
+
+## Run
+
+Create test namespace:
+```bash
+NAMESPACE=($(kubectl create -f ../namespace.yaml)[0])
+NAMESPACE=${NAMESPACE:10}
+```
+
+Register namespace in `spire` server:
+```bash
+kubectl exec -n spire spire-server-0 -- \
+/opt/spire/bin/spire-server entry create \
+-spiffeID spiffe://example.org/ns/${NAMESPACE}/sa/default \
+-parentID spiffe://example.org/ns/spire/sa/spire-agent \
+-selector k8s:ns:${NAMESPACE} \
+-selector k8s:sa:default
+```
+
+Select node to deploy NSC and NSE:
+```bash
+NODE=($(kubectl get nodes -o go-template='{{range .items}}{{ if not .spec.taints  }}{{index .metadata.labels "kubernetes.io/hostname"}} {{end}}{{end}}')[0])
+```
+
+Create Passthrough and firewall NSE configurations:
+```bash
+PASS_COUNT=4
+PASS_CFG=""
+NAME=""
+LABELS=""
+VOLUME_PATCH=""
+NS=""
+
+for ((i = 1; i <= PASS_COUNT; i++))
+do
+  f="passthrough-${i}"
+  NAME="nse-passthrough-${i}"
+  LABELS="app:passthrough-${i}"
+  if ((i == PASS_COUNT))
+  then
+    f="nse-firewall"
+    NAME="acl-filter"
+    LABELS="app:firewall"
+    VOLUME_PATCH="- config-patch.yaml"
+    NS="
+    - source_selector:
+        app: firewall${NS}
+      routes:
+        - destination_selector:
+            app: gateway
+    - routes:
+        - destination_selector:
+            app: firewall"
+  else
+    NS="${NS}
```

Comment:
Name of the variable is confusing.
Is this a variable for matches? If so, then why few lines above you use it in `app: firewall${NS}`?

Reply:
Please, examine this closely, it is creating exactly the same matches as previously have been in file for network service

Reply:
Yeah, now I see.
Well, as I said, name of this variable is confusing. Even though I already tried to look at the usage of this variable, I still had doubts until you confirmed that everything is correct.
I believe the example would be much more readable if we renamed this var to `MATCHES`.

Reply:
done